roofit/roofitcore/src/RooProduct.cxx

```diff
@@ -493,9 +493,21 @@ void RooProduct::setCacheAndTrackHints(RooArgSet& trackNodes)
   }
 }
 
+////////////////////////////////////////////////////////////////////////////////
 
-
-
+std::string RooProduct::translate(RooFit::Detail::CodeSquashContext &ctx) const
+{
+   std::string adResult;
+   // Build a (node1 * node2 * node3 * ...) like expression.
+   adResult = '(';
+   for (const auto item : _compRSet) {
+      auto rcomp = static_cast<RooAbsReal *>(item);
+      adResult += ctx.getResult(rcomp) + "*";
+   }
+   adResult.back() += ')';
+   ctx.addResult(this, adResult);
+   return "";
```

Comment:
```suggestion
   // Build a (node1 * node2 * node3 * ...) like expression.
   std::string result = "(";
   for (const auto item : _compRSet) {
      result += ctx.getResult(item) + "*";
   }
   result.back() = ')';
   ctx.addResult(this, result);
   return "";
```
The code can be simplified a bit, and it also had a bug. This is wrong: `adResult.back() += ')';`. Should be assignment, not increment. Also, the variable name should not contain `ad`.